service/history/shard/context.go

```diff
@@ -1194,10 +1194,25 @@ func (s *contextImpl) InsertFailoverMarkers(
 	defer s.updateMaxReadLevelLocked(transferMaxReadLevel)
 
 	var err error
-	for attempt := 0; attempt < conditionalRetryCount; attempt++ {
-		err = s.executionManager.CreateFailoverMarkerTasks(&persistence.CreateFailoverMarkersRequest{Markers: markers})
-		if err == nil {
+	for attempt := int32(0); attempt < conditionalRetryCount; attempt++ {
+		err = s.executionManager.CreateFailoverMarkerTasks(
+			&persistence.CreateFailoverMarkersRequest{
+				RangeID: s.getRangeID(),
+				Markers: markers,
+			},
+		)
+		switch err.(type) {
+		case nil:
 			break
+		case *persistence.ShardOwnershipLostError:
+			// do not retry on ShardOwnershipLostError
+			break
```

Comment:
`s.closeShard()` to trigger a shard close?

Reply:
good point. Updated